--- a/test/src/example.cpp
+++ b/test/src/example.cpp
@@ -9,8 +9,6 @@
 
 #include <godot_cpp/classes/global_constants.hpp>
 #include <godot_cpp/classes/label.hpp>
-#include <godot_cpp/classes/multiplayer_api.hpp>
-#include <godot_cpp/classes/multiplayer_peer.hpp>
 #include <godot_cpp/classes/os.hpp>
 #include <godot_cpp/variant/typed_dictionary.hpp>
 #include <godot_cpp/variant/utility_functions.hpp>
@@ -105,14 +103,6 @@
 }
 
 void Example::_notification(int p_what) {
-	if (p_what == NOTIFICATION_READY) {
-		Dictionary opts;
-		opts["rpc_mode"] = MultiplayerAPI::RPC_MODE_AUTHORITY;
-		opts["transfer_mode"] = MultiplayerPeer::TRANSFER_MODE_RELIABLE;
-		opts["call_local"] = true;
-		opts["channel"] = 0;
-		rpc_config("test_rpc", opts);
-	}
 	//UtilityFunctions::print("Notification: ", String::num(p_what));
 }
 
@@ -236,9 +226,6 @@
 
 	ClassDB::bind_method(D_METHOD("test_variant_iterator", "input"), &Example::test_variant_iterator);
 
-	ClassDB::bind_method(D_METHOD("test_rpc", "value"), &Example::test_rpc);
-	ClassDB::bind_method(D_METHOD("test_send_rpc", "value"), &Example::test_send_rpc);
-	ClassDB::bind_method(D_METHOD("return_last_rpc_arg"), &Example::return_last_rpc_arg);
 
 	ClassDB::bind_method(D_METHOD("def_args", "a", "b"), &Example::def_args, DEFVAL(100), DEFVAL(200));
 	ClassDB::bind_method(D_METHOD("callable_bind"), &Example::callable_bind);
@@ -659,18 +646,6 @@
 	return output;
 }
 
-void Example::test_rpc(int p_value) {
-	last_rpc_arg = p_value;
-}
-
-void Example::test_send_rpc(int p_value) {
-	rpc("test_rpc", p_value);
-}
-
-int Example::return_last_rpc_arg() {
-	return last_rpc_arg;
-}
-
 void Example::callable_bind() {
 	Callable c = Callable(this, "emit_custom_signal").bind("bound", 11);
 	c.call();
--- a/test/src/example.h
+++ b/test/src/example.h
@@ -83,7 +83,6 @@
 	Vector2 custom_position;
 	Vector3 property_from_list;
 	Vector2 dprop[3];
-	int last_rpc_arg = 0;
 
 	const bool object_instance_binding_set_by_parent_constructor;
 	bool has_object_instance_binding() const;
@@ -174,11 +173,6 @@
 
 	Variant test_variant_iterator(const Variant &p_input);
 
-	// RPC
-	void test_rpc(int p_value);
-	void test_send_rpc(int p_value);
-	int return_last_rpc_arg();
-
 	void callable_bind();
 
 	// Property.
--- a/test/project/main.gd
+++ b/test/project/main.gd
@@ -244,13 +244,6 @@
 	assert_equal(example.test_variant_iterator([10, 20, 30]), [15, 25, 35])
 	assert_equal(example.test_variant_iterator(null), "iter_init: not valid")
 
-	# RPCs.
-	assert_equal(example.return_last_rpc_arg(), 0)
-	example.test_rpc(42)
-	assert_equal(example.return_last_rpc_arg(), 42)
-	example.test_send_rpc(100)
-	assert_equal(example.return_last_rpc_arg(), 100)
-
 	# Virtual method.
 	var event = InputEventKey.new()
 	event.key_label = KEY_H
